okio/src/commonMain/kotlin/okio/Options.kt

```diff
@@ -40,7 +40,7 @@ class Options private constructor(
       // indexes to the caller's indexes.
       val list = byteStrings.toMutableList()
       list.sort()
-      val indexes = mutableListOf(*byteStrings.map { -1 }.toTypedArray())
+      val indexes = MutableList(list.size) { -1 }
```

Comment:
Yeah I didn't want to get too invasive. You would have to track the working length separately and then do a shift to simulate the `removeAt`. At some point it's diminishing returns since this function is only meant to be called once and then the resulting object cached for long periods of time.